src/mbgl/gl/gl_functions.hpp

```diff
@@ -0,0 +1,189 @@
+#pragma once
+
+#include <mbgl/gl/gl_defines.hpp>
+
+namespace mbgl {
+namespace gl {
+
+/**
+ * @brief Holds pointers to OpenGL ES 2.0 functions. Mapbox GL Native
+ * will only use OpenGL through this abstraction to make sure no assumption
+ * is made on how OpenGL is loaded by the platform and what OpenGL
+ * implementation is provided, as long as it is compatible with ES 2.0.
+ *
+ * Heavily inspired by QOpenGLFunctions, the recommend way of using this
+ * class is by protected/private inheritance, otherwise, the user of
```

Comment:
Could this class be be instead a singleton instance of `GLFunctionPointers` ?

Reply:
I will get rid of `GLFunctionPointers`, we won't need to instantiate anything.